actors.md

```diff
@@ -324,13 +324,35 @@ func SlashConsensusFault(block1, block2 BlockHeader) {
 		Fatal("blocks do not prove a slashable offense")
 	}
 
+	if !self.Miners.Has(msg.From) {
+		Fatal("slash collateral must only be called by a miner actor")
+	}
+
 	miner := AuthorOf(block1)
 
-	// TODO: Some of the slashed collateral should be paid to the slasher
+	// Check if miner has been slashed
+	if !self.Miners.Has(miner) {
```

Comment:
a miner is removed from the pool on slashing? this surprises me.

Reply:
Yes, this is what it currently is in the spec but likely to be changed. cc @whyrusleeping

Reply:
For consensus slashing, yes. the miner is removed from the set of valid miners.

Reply:
for storage fault slashing however, this should not be the case

Reply:
@decentralion since we are slashing for consensus fault, a miner is removed from the pool in this case.